messaging/src/main/java/org/axonframework/serialization/AbstractXStreamSerializer.java

```diff
@@ -65,6 +65,10 @@ protected AbstractXStreamSerializer(Builder builder) {
         this.converter = builder.converter;
         this.revisionResolver = builder.revisionResolver;
 
+        if (builder.classLoader != null) {
```

Comment:
Small pointer, but I feel it would be nicer to perform this validation and check in the `build()` method on the `XStreamSerializer`. Mainly because the `classLoader` doesn't mean anything from the perspective of this constructor, it's purely there to define what the `XStream` instance would look like.

Reply:
Probably this `if` block better move to `AbstractXStreamSerializer.Builder#validate` method close to this `XStream` configuration code:
```java
            if (lenientDeserialization) {
                xStream.ignoreUnknownElements();
            }
```
What do you think @smcvb ?

Reply:
Good call @aupodogov, sounds reasonable to place it there to me too!

Reply:
OK, done